react/shared.w

```diff
@@ -0,0 +1,51 @@
+bring cloud;
+bring fs;
+
+bring "./utils.w" as utils;
+
+pub struct AppPros {
+  projectPath: str;
+  localPort: num?;
+  buildDir: str?;
+  startCommand: str?;
+  buildCommand: str?;
+  useBuildCommand: bool?;
+}
+
+pub interface IApp {
+  getUrl(): str;
+  addEnvironment(key: str, value: str): void;
+}
+
+pub class AppBase {
+  protected props: AppPros;
+
+  protected path: str;
+
+  protected env: MutMap<str>;
+
+  protected startCommand: str;
+  protected buildCommand: str;
+  protected buildDir: str;
+
+  new(props: AppPros) {
+    this.path = fs.absolute(nodeof(this).app.entrypointDir, props.projectPath);
```

Comment:
@Chriscbr this is going to bite us in the A because we don't have something like `__dirname` yet.

So for example, if you create a library or a module in a subdirectory and want to put your project in an adjacent directory, this will not work.